tx-pool/src/component/orphan.rs

```diff
@@ -122,6 +123,15 @@ impl OrphanPool {
         if self.entries.contains_key(&tx.proposal_short_id()) {
             return;
         }
+
+        // double spend checking
+        if tx
```

Comment:
Do we need to be concerned about double-spending issues in the orphan pool?

Reply:
I discussed with @zhangsoledad , he mentioned one reason that `find_orphan_by_previous` return only one `OrphanEntry` is `orphan pool` may contain conflicted `tx` and suggest add double-spending check.

After txpool refactor, we have double-spending check when a `tx` is added to `Pending`, I think it is trivial we add it here. One result I can think of is that conflicted `tx` may broadcasted more.

Do you have more concerns about this?